packages/devtools_app/lib/src/performance/flutter/performance_screen.dart

```diff
@@ -29,82 +40,105 @@ class PerformanceScreen extends Screen {
   }
 }
 
-class PerformanceBody extends StatefulWidget {
-  const PerformanceBody();
+class PerformanceScreenBody extends StatelessWidget {
+  final PerformanceController controller = PerformanceController();
 
   @override
-  PerformanceBodyState createState() => PerformanceBodyState();
-}
-
-class PerformanceBodyState extends State<PerformanceBody> {
-  final PerformanceController _controller = PerformanceController();
-
-  CpuProfileData _data;
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            _buildStateControls(),
+            ProfileGranularityDropdown(),
+          ],
+        ),
+        Expanded(
+          child: ValueListenableBuilder(
+            valueListenable: controller.cpuProfilerController.dataNotifier,
+            builder: (context, cpuProfileData, _) {
+              if (cpuProfileData ==
+                  controller.cpuProfilerController.dummyEmptyCpuProfileData) {
```

Comment:
Nit: if this object is a special one that indicates a recording is in progress, I suggest naming it to reflect that.

Reply:
renamed to baseStateCpuProfileData to better reflect its purpose